tsdb/shard.go

```diff
@@ -173,6 +181,8 @@ func (s *Shard) Open() error {
 		}
 		s.logger.Printf("%s database index loaded in %s", s.path, time.Now().Sub(start))
 
+		go s.monitorSize()
```

Comment:
do we have a separate goroutine for monitoring each individual stat? Seems like we should come up with a design where on goroutine gathers all stats and just pulls them off some sort of buffer that everything sends them to?